runtime/protocol/aws-json-protocols/common/src/aws/smithy/kotlin/runtime/awsprotocol/json/RestJsonErrorDeserializer.kt

```diff
@@ -65,7 +67,7 @@ public object RestJsonErrorDeserializer {
                 }
             }
         }
-        return ErrorDetails(sanitize(code), message, requestId = null)
+        return ErrorDetails(chooseCode(headerCode, bodyCode, bodyType), message, requestId = null)
```

Comment:
**nit**: personally I wouldn't make a new function for this, a comment should be sufficient. interested in what other people think.
```kt
return ErrorDetails(sanitize(headerCode ?: bodyCode ?: bodyType), message, requestId = null)
```